text/0047-unspecified-semantics.md

```diff
@@ -0,0 +1,178 @@
+# Make `==` and `in` on `Unspecified` to be errors
+
+## Related issues and PRs
+
+- Reference Issues: [cedar#217](https://github.com/cedar-policy/cedar/issues/217)
+- Implementation PR(s):
+
+## Timeline
+
+- Started: 2024-02-05
+
+## Summary
+
+The `==` and `in` operations will always produce errors when applied to
+`Unspecified` entities. This replaces the current behavior, which produces
+errors in some cases, `true` in other cases, and `false` in other cases.
+
+## Basic example
+
+Cedar's API allows you to leave the principal, action, or resource of a request
+unspecified, indicating that the principal, action, or resource "shouldn't
+matter" for any policy evaluation.
+Suppose `principal` is left unspecified in a request.
+
+1. If some policy contains the expression `principal == User::"alice"` or
+`User::"alice" == principal`, those expressions currently (Cedar 2.x and 3.x)
+evaluate to `false` -- unspecified entities are defined to not be equal to any
+entity in the hierarchy.
+2. If some policy contains the expression `principal in Group::"foo"` or
+`User::"alice" in principal`, those expressions currently (Cedar 2.x and 3.x)
+evaluate to `false` -- unspecified entities are defined to not be `in` any
+entity in the hierarchy.
+3. If some policy contains the expression `principal == principal` or
+`principal in principal`, those expressions currently (Cedar 2.x and 3.x)
+evaluate to `true` (even when `principal` is unspecified in the request).
+Unspecified entities are defined to be equal to themselves.
+4. If some policy contains the expression `principal has foo`, that expression
+currently (Cedar 2.x and 3.x) evaluates to `false` -- unspecified entities are
+defined to have no attributes.
+4. If some policy contains the expression `principal.foo`, that expression
+currently throws an error.
+
+With this RFC, we change the behavior to be consistent: _All_ of the above
+examples will throw an error.
```

Comment:
I presume the use of `is` on an unspecified entity also results in an error. Any other operators to mention?

Reply:
Today, `is` always returns `false` for unspecified entities.  This RFC proposes an error.  I will adjust the RFC text, good catch